### PR TITLE
Enable nginx HTTP compression

### PR DESCRIPTION
**Pull request by oncletom**
The pull request is composed of 2 commits:

1. one with the initial configuration, found on the production server.
2. one with compression, security and 404 page settings.

I have slightly reworded the `development` config file so as the routes (`/static` and `/api`) match `production` ones.

### CODE REVIEW

File: infrastructure/nginx/sites-available/development [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+
+	error_page 404 error_404.html
+
+	# Security
+	add_header X-XSS-Protection "1; mode=block" always;
+	add_header X-Content-Type-Options nosniff always;
+	add_header X-Frame-Options SAMEORIGIN always;
+
+	# Cross Domain API
+	add_header "Access-Control-Allow-Origin" "*";
```
Comment (daniviga) [inline infrastructure/nginx/sites-available/development]: Are you sure you want to enable CORS to allow access from any host? The API is under `/api` (same domain) so it is not affected by CORS (unless there's a scheme mismatch between client and API requests)

Comment (daniviga) [inline infrastructure/nginx/sites-available/development]: Are you sure you want to enable CORS to allow access from any host? The API is under /api (same domain) so it is not affected by CORS (unless there's a scheme mismatch between client and API requests).

File: infrastructure/nginx/sites-available/production [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+	error_page 404 error_404.html
+
+	# Security
+	add_header X-XSS-Protection "1; mode=block" always;
+	add_header X-Content-Type-Options nosniff always;
+	add_header X-Frame-Options SAMEORIGIN always;
+
+	# Cross Domain API
+	add_header "Access-Control-Allow-Origin" "*";
```
Comment (daniviga) [inline infrastructure/nginx/sites-available/production]: As above. At least in production I would enforce the same-origin policy (at least for `POST` and `PUT`).

Comment (oncletom) [reply infrastructure/nginx/sites-available/production]: Yes you're right, it does not matter to be cross-origin in development. I will remove this line.

File: infrastructure/nginx/sites-available/development (hunk 1 above)
Comment (oncletom) [reply infrastructure/nginx/sites-available/development]: Yes you're right, it does not matter to be cross-origin in development. I will remove this line.

Comment (nastasi-oq) [reply infrastructure/nginx/sites-available/development]: The CORS header is still there, some other reason to keep it there ?

Comment (oncletom) [reply infrastructure/nginx/sites-available/development]: @nastasi-oq yes, I have not made the change yet 🙂

Comment (oncletom) [reply infrastructure/nginx/sites-available/development]: Done 👍

File: infrastructure/nginx/sites-available/production (hunk 2 above)
Comment (oncletom) [reply infrastructure/nginx/sites-available/production]: Done 👍

**Comment by oncletom** [comment]
Perfect. Feel free to close this issue once the configurations have been applied.

I have committed these files in an `infrastructure/` folder so as the `infrastructure/nginx/sites-available` folder can be clones and symlinked as `/etc/nginx/sites-available` to ease up a shared work on the configuration.

**Comment by nastasi-oq** [comment]
@oncletom I opened a PR on your ``dtc-innovation:feature/infra-config`` fork with some change in the ``development`` configuration.

**Comment by oncletom** [comment]
@nastasi-oq ah I see! I did not realise it was a PR. I cherry picked the commit to add it to the branch directly so it should be good to go :-)

**Comment by oncletom** [comment]
@nastasi-oq understood — what I meant is your changes are now part of this pull request as well (you should be able to see your commit here).

**Comment by nastasi-oq** [comment]
> @nastasi-oq understood — what I meant is your changes are now part of this pull request as well (you should be able to see your commit here).

I understood, I mean that there are some new commits to integrate

**Comment by nastasi-oq** [comment]
@oncletom should we merge and start the integration on dev machine ?

**Comment by oncletom** [comment]
@nastasi-oq yes please 🙂

**Pull request merged**